src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2577,7 +2577,13 @@ int RedirectedHandledJITCaseExceptionFilter(
     pFrame->Pop();
 
     // Copy the saved context record into the EH context;
-    ReplaceExceptionContextRecord(pExcepPtrs->ContextRecord, pCtx);
+    // NB: cannot use ReplaceExceptionContextRecord here.
+    //     these contexts may contain extended registers and may have different format
+    //     for reasons such as alignment or context compaction
+    //
+    // REVIEW: CopyContext may fail. in theory. How should we handle that? FailFast?
```

Comment:
Yes, FailFast is appropriate for failure here.

Reply:
However, be careful calling new functions from this function. If they have destructors and the compiler chooses to inline them, the runtime will start crashing.

Reply:
I was thinking of code like:
```c
            if (!CopyContext(pTarget, pTarget->ContextFlags, pCtx))
            {
                STRESS_LOG1(LF_EH, LL_ERROR, "ERROR: Could not set context record, lastError = 0x%x\n", GetLastError());
                ThrowLastError();
            }
```

Would that fit here?

It is highly unlikely that copy would fail, but it'd be nice to have some diagnostics if that start happenning.